permission denied in clone

**What happened**:
复现步骤：
```
root@bench-03:~/juicefs# cat .github/scripts/hypo/command_test.py
    def test_clone_4834(self):
        state = JuicefsCommandMachine()
        folders_0 = state.init_folders()
        state.chmod(entry=folders_0, mode=2427, user='root')
        folders_1 = state.mkdir(mode=2931, parent=folders_0, subdir='vhjp', umask=369, user='root')
        state.chmod(entry=folders_1, mode=1263, user='root')
        state.clone(entry=folders_1, new_entry_name='tbim', parent=folders_0, preserve=False, user='user1')
        state.teardown()
root@bench-03:~/juicefs# ./juicefs format redis://localhost/1 redis --enable-acl --trash-days 0
root@bench-03:~/juicefs# ./juicefs mount redis://localhost/1 /tmp/jfs1 -d
root@bench-03:~/juicefs# ./juicefs format mysql://juicedata:juicedata@(127.0.0.1)/test mysql --enable-acl --trash-days 0
root@bench-03:~/juicefs# ./juicefs mount mysql://juicedata:juicedata@(127.0.0.1)/test /tmp/jfs2 -d
root@bench-03:~/juicefs#  python3 .github/scripts/hypo/command_test.py -k test_clone
```


```
2024-05-10 18:10:30,221 - INFO - run_cmd: sudo -u user1 ./juicefs clone /tmp/jfs1/fsrand/vhjp /tmp/jfs1/fsrand/tbim
2024-05-10 18:10:30,552 - INFO - do_clone succeed
2024-05-10 18:10:30,552 - INFO - run_cmd: sudo -u user1 ./juicefs clone /tmp/jfs2/fsrand/vhjp /tmp/jfs2/fsrand/tbim
2024-05-10 18:10:30,882 - INFO - do_clone  {} failed:  clone failed: permission denied

F
======================================================================
FAIL: test_clone2 (__main__.TestCommand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".github/scripts/hypo/command_test.py", line 65, in test_clone2
    state.clone(entry=folders_1, new_entry_name='tbim', parent=folders_0, preserve=False, user='user1')
  File "/root/juicefs/.github/scripts/hypo/command.py", line 155, in clone
    entry = Entries.filter(lambda x: x != multiple()),
  File "/usr/local/lib/python3.8/dist-packages/hypothesis/stateful.py", line 734, in rule_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/command.py", line 155, in clone
    entry = Entries.filter(lambda x: x != multiple()),
  File "/usr/local/lib/python3.8/dist-packages/hypothesis/stateful.py", line 855, in precondition_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/command.py", line 165, in clone
    assert self.equal(result1, result2), f'\033[31mclone:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
AssertionError: clone:
result1 is True
result2 is  clone failed: permission denied
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/redis.go
+++ b/pkg/meta/redis.go
@@ -4135,14 +4135,15 @@ func (m *redisMeta) LoadMeta(r io.Reader) (err error) {
 	return err
 }
 
-func (m *redisMeta) doCloneEntry(ctx Context, srcIno Ino, parent Ino, name string, ino Ino, attr *Attr, cmode uint8, cumask uint16, top bool) syscall.Errno {
+func (m *redisMeta) doCloneEntry(ctx Context, srcIno Ino, parent Ino, name string, ino Ino, originAttr *Attr, cmode uint8, cumask uint16, top bool) syscall.Errno {
 	return errno(m.txn(ctx, func(tx *redis.Tx) error {
 		a, err := tx.Get(ctx, m.inodeKey(srcIno)).Bytes()
 		if err != nil {
 			return err
 		}
-		m.parseAttr(a, attr)
-		if eno := m.Access(ctx, srcIno, MODE_MASK_R, attr); eno != 0 {
+		m.parseAttr(a, originAttr)
+		attr := *originAttr
+		if eno := m.Access(ctx, srcIno, MODE_MASK_R, &attr); eno != 0 {
 			return eno
 		}
 		attr.Parent = parent
@@ -4191,7 +4192,7 @@ func (m *redisMeta) doCloneEntry(ctx Context, srcIno Ino, parent Ino, name strin
 		}
 
 		_, err = tx.TxPipelined(ctx, func(p redis.Pipeliner) error {
-			p.Set(ctx, m.inodeKey(ino), m.marshal(attr), 0)
+			p.Set(ctx, m.inodeKey(ino), m.marshal(&attr), 0)
 			p.IncrBy(ctx, m.usedSpaceKey(), align4K(attr.Length))
 			p.Incr(ctx, m.totalInodesKey())
 			if len(srcXattr) > 0 {

--- a/pkg/meta/tkv.go
+++ b/pkg/meta/tkv.go
@@ -3398,14 +3398,15 @@ func (m *kvMeta) LoadMeta(r io.Reader) error {
 	})
 }
 
-func (m *kvMeta) doCloneEntry(ctx Context, srcIno Ino, parent Ino, name string, ino Ino, attr *Attr, cmode uint8, cumask uint16, top bool) syscall.Errno {
+func (m *kvMeta) doCloneEntry(ctx Context, srcIno Ino, parent Ino, name string, ino Ino, originAttr *Attr, cmode uint8, cumask uint16, top bool) syscall.Errno {
 	return errno(m.txn(func(tx *kvTxn) error {
 		a := tx.get(m.inodeKey(srcIno))
 		if a == nil {
 			return syscall.ENOENT
 		}
-		m.parseAttr(a, attr)
-		if eno := m.Access(ctx, srcIno, MODE_MASK_R, attr); eno != 0 {
+		m.parseAttr(a, originAttr)
+		attr := *originAttr
+		if eno := m.Access(ctx, srcIno, MODE_MASK_R, &attr); eno != 0 {
 			return eno
 		}
 		attr.Parent = parent
@@ -3455,7 +3456,7 @@ func (m *kvMeta) doCloneEntry(ctx Context, srcIno Ino, parent Ino, name string, 
 			}
 		}
 
-		tx.set(m.inodeKey(ino), m.marshal(attr))
+		tx.set(m.inodeKey(ino), m.marshal(&attr))
 		prefix := m.xattrKey(srcIno, "")
 		tx.scan(prefix, nextKey(prefix), false, func(k, v []byte) bool {
 			tx.set(m.xattrKey(ino, string(k[len(prefix):])), v)

